### PR TITLE
EL-704 upper tribunal certificated thresholds

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/EL-704

Changes to thresholds for upper tribunal immigration certificated cases

### CODE REVIEW

**Comment by patrick-laa** [comment]
@starswan I'm going to hold off on reviewing this for a sec because there's a change needed to EL-695 (#84) that came to light on [a Jira comment on EL-698](https://dsdmoj.atlassian.net/browse/EL-698?focusedCommentId=270951) that I think will change this PR quite a bit. It turns out the actual logic is:

- Ignore whether it's certificated or controlled. If it's an Immigration case (Asylum doesn't count), use a special capital upper threshold of £3k. Otherwise change nothing.

File: app/services/creators/capital_eligibility_creator.rb [hunk 1]
```diff
@@ -25,6 +25,20 @@ def create_eligibility(ptc)
           lower_threshold: controlled_threshold(ptc),
           assessment_result: "pending",
         )
+      elsif ptc.to_sym == CFEConstants::IMMIGRATION_PROCEEDING_TYPE_CCMS_CODE
```
Comment (patrick-laa) [inline app/services/creators/capital_eligibility_creator.rb:28]: I think we're making this class a lot more complicated than we need to. We've got a 4-block if/else statement here and another if block inside `controlled_threshold`. The problem is that we're splitting up the logic by level of help first and proceeding type second, but actually the proceeding type affects both controlled and certificated work in the same way. And there's only one special case, which is Immigration. So I think you can safely drop the whole `ptc.to_sym == CFEConstants::ASYLUM_PROCEEDING_TYPE_CCMS_CODE` branch, and if you pull out the `ptc.to_sym == CFEConstants::IMMIGRATION_PROCEEDING_TYPE_CCMS_CODE` above the check for controlled/certificated you can get that out of the way in one fell swoop, and make this whole class much less complex.

Comment (patrick-laa) [reply app/services/creators/capital_eligibility_creator.rb:28]: Although I've missed something in the above: lower limits not applying in for immigration/asylum proceeding types for certificated cases. So maybe it's not quite as neat as I thought.

Comment (starswan) [reply app/services/creators/capital_eligibility_creator.rb:27]: Yes I've been grappling with the clearest way to code what is essentially a very simple rule!

**Review by patrick-laa — APPROVED** [review]

**Pull request merged**